hyperspy/_signals/lazy.py

```diff
@@ -634,154 +635,193 @@ def _block_iterator(self,
             The navigation locations marked as True are not used in the
             decompostion.
         signal_mask : {BaseSignal, numpy array, dask array}
             The signal locations marked as True are not used in the
             decomposition.
-        reproject : bool
+        reproject : bool, default True
             Reproject data on the learnt components (factors) after learning.
+        print_info : bool, default True
+            If True, print information about the decomposition being performed.
+            In the case of sklearn.decomposition objects, this includes the
+            values of all arguments of the chosen sklearn algorithm.
         **kwargs
             passed to the partial_fit/fit functions.
 
-        Notes
-        -----
-        Various algorithm parameters and their default values:
-            ORNMF:
-                lambda1=1,
-                kappa=1,
-                robust=False,
-                store_r=False
-                batch_size=None
-            ORPCA:
-                fast=True,
-                lambda1=None,
-                lambda2=None,
-                method=None,
-                learning_rate=None,
-                init=None,
-                training_samples=None,
-                momentum=None
-            PCA:
-                batch_size=None,
-                copy=True,
-                white=False
-
+        References
+        ----------
+        .. [KeenanKotula2004] M. Keenan and P. Kotula, "Accounting for Poisson noise
+            in the multivariate analysis of ToF-SIMS spectrum images", Surf.
+            Interface Anal 36(3) (2004): 203-212.
+
+        See Also
+        --------
+        * :py:meth:`~.learn.mva.MVA.decomposition` for non-lazy signals
+        * :py:func:`dask.array.linalg.svd`
+        * :py:class:`sklearn.decomposition.IncrementalPCA`
+        * :py:class:`~.learn.rpca.ORPCA`
+        * :py:class:`~.learn.ornmf.ORNMF`
 
         """
-        if bounds:
-            msg = (
+        if kwargs.get("bounds", False):
+            warnings.warn(
                 "The `bounds` keyword is deprecated and will be removed "
-                "in v2.0. Since version > 1.3 this has no effect.")
-            warnings.warn(msg, VisibleDeprecationWarning)
+                "in v2.0. Since version > 1.3 this has no effect.",
+                VisibleDeprecationWarning,
+            )
+            kwargs.pop("bounds", None)
+
+        # Deprecate 'ONMF' for 'ORNMF'
+        if algorithm == "ONMF":
+            warnings.warn(
+                "The argument `algorithm='ONMF'` has been deprecated and may "
+                "be removed in future. Please use `algorithm='ornmf'` instead.",
+                VisibleDeprecationWarning,
+            )
+            algorithm = "ornmf"
+
+        # Deprecate uppercase to favour lowercase (consistent
+        # with non-lazy decomposition)
+        if algorithm in ["PCA", "ORPCA", "ORNMF"]:
```

Comment:
I can do that.

Reply:
Actually, @francisco-dlp, I chose lowercase because that's how sklearn typically takes all of its string arguments. I could also do `.lower()` and avoid the deprecation altogether.

Reply:
On reflection, I'll take your approach and go with uppercase. Working on it now.

Reply:
Since we can actually deprecate "sklearn_pca" and "sklearn_fastica" etc in future as a result of this pull request (and get the user to pass their own objects), the lowercase no longer matters.

Reply:
I've left them as they are for now, and made all the other changes @francisco-dlp.